config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/delete' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/update' => 'references#edit_abstract', as: :edit_abstract
```

Comment:
Instead of `get` it should say `patch` or `update` or both and the action should just be called `#edit`, unless you're actually updating the record, then it should be `update`